airbyte-integrations/connectors/source-bigquery/source_bigquery/streams.py

```diff
@@ -327,95 +557,165 @@ def _updated_state(self, current_stream_state: MutableMapping[str, Any], latest_
         """
         if not self.cursor_field:
             return self.state
-        
         latest_record_state = latest_record[self.cursor_field]
         stream_state = current_stream_state.get(self.cursor_field, None)
-
         if stream_state:
             self._cursor = max(latest_record_state, stream_state)
         else:
             self._cursor = latest_record_state
-        self.state = {self.cursor_field: self._cursor} 
-
+        self.state = {self.cursor_field: self._cursor}
         if datetime.now() >= self._checkpoint_time + timedelta(minutes=15):
             self.checkpoint(self.name, self.state, self.namespace)
             self._checkpoint_time = datetime.now()
-
         return self.state
 
+    def _extract_borders(self):
+        if not self.start_time and not self.end_time:
```

Comment:
Why are `self.start_time` and `self.end_time` instance variables? It seems like they could be local variables to this method which would remove some of the statefulness of this class

Reply:
I made them instance variables so that I only set them once since I do and http request to set their value, I thought it will be best if we only did this once

Reply:
Why can't we deduce those values from the state?

Reply:
Answered here https://github.com/artefactory/airbyte/pull/22/files#r1642315045